plugin/activitywatch.vim

```diff
@@ -105,6 +107,20 @@ function! s:CreateBucket()
     call HTTPPostJson(s:bucket_apiurl, l:body)
 endfunc
 
+function! s:GetGitBranch(localtime)
+    if a:localtime - s:last_branch_update > 5
+        let l:cmd_result = systemlist('git branch --show-current')[0]
```

Comment:
This call is still synchronous. So while it's an improvement, there's still a risk of VIM freezing when doing s blocking operation like systemlist. It would be optimal if we did something similar as in HTTPPostJson with jobstart.